src/EditorFeatures/CSharpTest/ImplementInterface/ImplementInterfaceTests.cs

```diff
@@ -6543,43 +6605,41 @@ partial class C
 }}", index: 3);
         }
 
-        private static string DisposePattern(string disposeVisibility, string className, string implementationVisibility)
+        private static string DisposePattern(
+            string disposeVisibility,
+            string className,
+            string implementationVisibility,
+            string disposeField = "disposedValue",
+            string gcPrefix = "")
         {
-            return $@"    #region IDisposable Support
-    private bool disposedValue = false; // {FeaturesResources.To_detect_redundant_calls}
-
-    {disposeVisibility}void Dispose(bool disposing)
+            return $@"    {disposeVisibility}void Dispose(bool disposing)
     {{
-        if (!disposedValue)
+        if (!{disposeField})
         {{
             if (disposing)
             {{
                 // {FeaturesResources.TODO_colon_dispose_managed_state_managed_objects}
             }}
 
-            // {CSharpFeaturesResources.TODO_colon_free_unmanaged_resources_unmanaged_objects_and_override_a_finalizer_below}
+            // {FeaturesResources.TODO_colon_free_unmanaged_resources_unmanaged_objects_and_override_finalizer}
             // {FeaturesResources.TODO_colon_set_large_fields_to_null}
-
-            disposedValue = true;
+            {disposeField} = true;
         }}
     }}
 
-    // {CSharpFeaturesResources.TODO_colon_override_a_finalizer_only_if_Dispose_bool_disposing_above_has_code_to_free_unmanaged_resources}
+    // // {string.Format(FeaturesResources.TODO_colon_override_finalizer_only_if_0_has_code_to_free_unmanaged_resources, "Dispose(bool disposing)")}
     // ~{className}()
     // {{
-    //   // {CSharpFeaturesResources.Do_not_change_this_code_Put_cleanup_code_in_Dispose_bool_disposing_above}
-    //   Dispose(false);
+    //     // {string.Format(FeaturesResources.Do_not_change_this_code_Put_cleanup_code_in_0_method, "Dispose(bool disposing)")}
+    //     Dispose(disposing: false);
     // }}
 
-    // {CSharpFeaturesResources.This_code_added_to_correctly_implement_the_disposable_pattern}
     {implementationVisibility}Dispose()
     {{
-        // {CSharpFeaturesResources.Do_not_change_this_code_Put_cleanup_code_in_Dispose_bool_disposing_above}
-        Dispose(true);
-        // {CSharpFeaturesResources.TODO_colon_uncomment_the_following_line_if_the_finalizer_is_overridden_above}
-        // GC.SuppressFinalize(this);
-    }}
-    #endregion";
+        // {string.Format(FeaturesResources.Do_not_change_this_code_Put_cleanup_code_in_0_method, "Dispose(bool disposing)")}
+        Dispose(disposing: true);
```

Comment:
💭 I'm not a _huge_ fan of making this a named argument, but I'm ignoring it for now since the long-term goal is to remove the `Dispose(bool)` method altogether.